src/orange/OrangeTrackView.hh

```diff
@@ -251,6 +269,51 @@ OrangeTrackView& OrangeTrackView::operator=(DetailedInitializer const& init)
     return *this;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * The current position.
+ */
+CELER_FUNCTION Real3 const& OrangeTrackView::pos() const
+{
+    return this->make_lsa().pos();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * The current direction.
+ */
+CELER_FUNCTION Real3 const& OrangeTrackView::dir() const
+{
+    return this->make_lsa().dir();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * The current volume ID (null if outside).
+ */
+CELER_FUNCTION VolumeId OrangeTrackView::volume_id() const
+{
+    return this->make_lsa().vol();
```

Comment:
We might want to think about having separate types for local and global volume/surface IDs. For local tracking we need the local volume ID (which is sort of an "implementation detail") but this function is part of the interface to the rest of the code and should therefore be the global volume ID of the lowest-level volume.

Reply:
I agree --- I will implement in the next PR.